src/sdk/pynni/nni/counter.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import torch
+import torch.nn as nn
+from torch.nn.modules.conv import _ConvNd
+from torch.nn.modules.batchnorm import _BatchNorm
+from nni.compression.torch.compressor import PrunerModuleWrapper
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from thop import profile
+except ImportError:
+    _logger.warning('Please install thop first.')
+    
+def count_flops_params(model: nn.Module, input_size=None, verbose=True):
+    """
+    Count FLOPs and Params of the given model. 
+    This function would identify the mask on the module 
+    and take the pruend shape into consideration. 
+    Note that, we only identify the output shape of the module, 
+    so the calculated FLOPs will be larger than real number.
+
+    Parameters
+    ---------
+    model : nn.Module
+        target model.
+    input_size: list, tuple
+        the input shape of data
+
+
+    Returns
+    -------
+    flops: float
+        total flops of the model
+    params:
+        total params of the model
+    """
+
+    assert input_size is not None
+
+    device = next(model.parameters()).device
+    inputs = torch.randn(input_size).to(device)
+
+    hook_module_list = []
+    prev_m = None
+    for m in model.modules():
+        weight_mask = None
+        m_type = type(m)
+        if m_type in custom_ops:
+            if isinstance(prev_m, PrunerModuleWrapper):
+                weight_mask = prev_m.weight_mask
+
+            m.register_buffer('weight_mask', weight_mask)
```

Comment:
no need to register buffer here if the model is not `PrunerModuleWrapper`?

Reply:
We do not directly register buffer on `PrunerModuleWrapper`. We only register buffer on `Conv` or `Linear` (please see custom_ops in Line 96) where the previous module was `PrunerModuleWrapper`.